### ISSUE TITLE
Setup does not work for newer MSS310

### ISSUE THREAD

**Issue by fritut08** [bytespider/Meross#72]
I used your software for pairing a few Meross Smart Plugs (MSS310, hardware 6.0.0, firmware 6.3.21) with my local mqtt broker a year ago which worked just fine. Now, I bought another set of those plugs (MSS310, hardware 8.0.0, firmware 8.3.4, not compatible with HomeKit or Matter) but am running into all kinds of problems during the pairing process. First of all, the new plugs do not seem to create a wifi network to connect to. This should not be a problem, though, since I can connect the plug to my wifi using the meross app. ( I do not know that works without the wifi network, though.) I then get the IP address of the plug from my router. My Home Assistant automatically recognizes the smart plug and shows me its key. I then use the following commands to get information from the plug and connect it to the MQTT broker.

```
./meross-info --verbose --include-wifi --key 'KEY_RETRIEVED_FROM_HOMEASSISTANT'
```
```
./meross-setup --gateway 'IP_RETRIEVED_FROM_ROUTER' --wifi-ssid "MY_WIFI_SSID" --wifi-pass 'MY_WIFI_PASSWD' --mqtt mqtts://IP_ADDRESS_OF_MQTT_BROKER:PORT --wifi-cipher 3 --wifi-encryption 6 --wifi-channel 6 --verbose --key 'KEY_RETRIEVED_FROM_HOMEASSISTANT'
```

The output of the last command shows the correct IP and port in a table and the correctly base64 encoded ssid and password of my wifi in a message. However, after the device reboots, it cannot connect to the wifi network anymore. Its LED alternates between yellow and green.

**Comment by bytespider** [comment]
Is there any information in the verbose output that would be helpful?

Also you used double quotes on your WiFi ssid, so try replacing with single quotes.

I wonder if the device ssid is hidden

**Comment by bytespider** [comment]
if you want to try it, there's an experimental branch that I started which allows you to send only parts of the setup.
https://github.com/bytespider/Meross/tree/feature/rewrite

**Comment by M0NsTeRRR** [comment]
Hello, tested today with two MSS315, firmware 9.5.31 and hardware 9.0.0. It seems that the SSID is hidden maybe it's the same for you @fritut08.

My setup with home assistant :
1) Install official meross app
2) Setup the plug with meross app (it will configure wifi)
3) On home assistant install https://github.com/krahabb/meross_lan, connect meross cloud account. Add devices and get key from one device.
3) Install bytespider project and comment out line line 95 to 102 (as we don't change wifi settings only MQTT) in `bin/meross-setup`
4) Get IP address of the plug from your DHCP and do `npx meross info --verbose -g <IP> --key <KEY>`, you should see plug configuration, it means everything is setup and you can go below (step not required)
5) Reconfigure meross MQTT plug `npx meross setup --gateway 'IP' --wifi-ssid 'wedontcare' --wifi-pass 'wedontcare' --mqtt mqtts://<IP or HOSTNAME of MQTT> --wifi-cipher 3 --wifi-encryption 6 --wifi-channel 6 --verbose --key 'KEY'`
6) Do step 4, it should display your new MQTT broker :)

PS : I've tried with the rewrite branch at beginning but I can't get it worked as their is works to do, old function not renammed like `setSystemTime()` to `configureSystemTime()`, missing an `await` keyword on `_send` function for `got` library and still after that I got a `ECONNRESET`. Much simpler to comment out wifi settings :)

Thanks for your library I can keep my privacy away from third parties cloud :)

**Comment by jluixjurado** [comment]
Hi there. I've got the same problem with my new mss310. What i've found is that new plugs doesn't seem to communicate through wifi but through bluetooth (in fact, when scanning for bluetooth devices you can find something like MRBL_XXXX, where XXXX are the last four digits of the plug MAC address. That's why CUSTOM PAIRER can't find the new plugs!

**Comment by bytespider** [comment]
@jluixjurado 
> What i've found is that new plugs doesn't seem to communicate through wifi but through bluetooth (in fact, when scanning for bluetooth devices you can find something like MRBL_XXXX, where XXXX are the last four digits of the plug MAC address.

That poses an issue of these new plugs. What I would recommend is to pair using the Meross app, then once it's on your network, reconfigure it using the tools.


@M0NsTeRRR 
> PS : I've tried with the rewrite branch at beginning but I can't get it worked as their is works to do

Sorry, I continued working on it and I pushed the changes because I needed to move to a new machine. I have time off now, so I'll endeavour to get this into a working state before New Year.

**Comment by M0NsTeRRR** [comment]
> @jluixjurado 
> > What i've found is that new plugs doesn't seem to communicate through wifi but through bluetooth (in fact, when scanning for bluetooth devices you can find something like MRBL_XXXX, where XXXX are the last four digits of the plug MAC address.
> 
> That poses an issue of these new plugs. What I would recommend is to pair using the Meross app, then once it's on your network, reconfigure it using the tools.
> 
> 
> @M0NsTeRRR 
> > PS : I've tried with the rewrite branch at beginning but I can't get it worked as their is works to do
> 
> Sorry, I continued working on it and I pushed the changes because I needed to move to a new machine. I have time off now, so I'll endeavour to get this into a working state before New Year.

Sure no worries, it's how open source works 😁. Thanks for your amazing jobs ;)

**Comment by fuomag9** [comment]
> @jluixjurado
> 
> > What i've found is that new plugs doesn't seem to communicate through wifi but through bluetooth (in fact, when scanning for bluetooth devices you can find something like MRBL_XXXX, where XXXX are the last four digits of the plug MAC address.
> 
> That poses an issue of these new plugs. What I would recommend is to pair using the Meross app, then once it's on your network, reconfigure it using the tools.
> 
> @M0NsTeRRR
> 
> > PS : I've tried with the rewrite branch at beginning but I can't get it worked as their is works to do
> 
> Sorry, I continued working on it and I pushed the changes because I needed to move to a new machine. I have time off now, so I'll endeavour to get this into a working state before New Year.

As referenced here https://github.com/krahabb/meross_lan/issues/346#issuecomment-1868544218 I've created a small pull request to allow for mqtt setup only, which fixes this issue https://github.com/bytespider/Meross/pull/75

**Comment by bytespider** [comment]
I have published a new release that includes #75
